source/transfers/test/TransferHandlerRegistry.js

```diff
@@ -0,0 +1,1008 @@
+      await tokenAST.transfer(aliceAddress, 1, { from: bobAddress })
+
+      // previous balances unchanged
+      ok(
+        await balances(aliceAddress, [
+          [tokenAST, 800],
+          [tokenDAI, 50],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenAST, 200],
+          [tokenDAI, 950],
+        ]),
+        'Bob balances are incorrect'
+      )
+
+      // increase allowances again
+      await tokenAST.approve(swapAddress, 50, { from: aliceAddress })
+      await tokenDAI.approve(swapAddress, 950, { from: bobAddress })
+
+      ok(
+        await allowances(aliceAddress, swapAddress, [
+          [tokenAST, 50],
+          [tokenDAI, 0],
+        ])
+      )
+      ok(
+        await allowances(bobAddress, swapAddress, [
+          [tokenAST, 0],
+          [tokenDAI, 950],
+        ])
+      )
+    })
+  })
+
+  describe('Swaps (Non-standard Fungible)', async () => {
+    let _order
+
+    before('Alice creates an order for Bob (200 OMG for 50 DAI)', async () => {
+      _order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenOMG.address,
+          amount: 200,
+        },
+        sender: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 50,
+        },
+      })
+
+      _order.signature = await signatures.getWeb3Signature(
+        _order,
+        aliceAddress,
+        swapAddress,
+        GANACHE_PROVIDER
+      )
+    })
+
+    it('Checks that Bob can swap with Alice (200 OMG for 50 DAI)', async () => {
+      emitted(await swap(_order, { from: bobAddress }), 'Swap')
+    })
+
+    it('Checks balances...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenOMG, 800],
+          [tokenDAI, 100],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenOMG, 200],
+          [tokenDAI, 900],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+
+    it('Checks that Bob cannot take the same order again (200 OMG for 50 DAI)', async () => {
+      await reverted(
+        swap(_order, { from: bobAddress }),
+        'ORDER_TAKEN_OR_CANCELLED'
+      )
+    })
+
+    it('Checks balances...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenOMG, 800],
+          [tokenDAI, 100],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenOMG, 200],
+          [tokenDAI, 900],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+
+    it('Checks that Alice cannot trade more than approved (200 OMG)', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenOMG.address,
+          amount: 200,
```

Comment:
again, add a full sender to this test, it currently has an address of 0x0 which would also fail the test